code/game/machinery/computer/prisoner.dm

```diff
@@ -14,89 +14,83 @@
 	var/stop = 0.0
 	var/screen = 0 // 0 - No Access Denied, 1 - Access allowed
 
+/obj/machinery/computer/prisoner/attack_ai(var/mob/user)
+	return src.attack_hand(user)
 
-	attack_ai(var/mob/user as mob)
-		return src.attack_hand(user)
-
-	attack_hand(var/mob/user as mob)
-		if(..())
-			return
-		user.set_machine(src)
-		var/dat
-		dat += "<B>Prisoner Implant Manager System</B><BR>"
-		if(screen == 0)
-			dat += "<HR><A href='?src=\ref[src];lock=1'>Unlock Console</A>"
-		else if(screen == 1)
-			dat += "<HR>Chemical Implants<BR>"
-			var/turf/Tr = null
-			for(var/obj/item/implant/chem/C in world)
-				Tr = get_turf(C)
-				if((Tr) && !AreConnectedZLevels(Tr.z, src.z))	continue // Out of range
-				if(!C.implanted) continue
-				dat += "[C.imp_in.name] | Remaining Units: [C.reagents.total_volume] | Inject: "
-				dat += "<A href='?src=\ref[src];inject1=\ref[C]'>(<font color=red>(1)</font>)</A>"
-				dat += "<A href='?src=\ref[src];inject5=\ref[C]'>(<font color=red>(5)</font>)</A>"
-				dat += "<A href='?src=\ref[src];inject10=\ref[C]'>(<font color=red>(10)</font>)</A><BR>"
-				dat += "********************************<BR>"
-			dat += "<HR>Tracking Implants<BR>"
-			for(var/obj/item/implant/tracking/T in world)
-				Tr = get_turf(T)
-				if((Tr) && !AreConnectedZLevels(Tr.z, src.z))	continue // Out of range
-				if(!T.implanted) continue
-				var/loc_display = "Space"
-				var/mob/living/carbon/M = T.imp_in
-				if(!istype(M.loc, /turf/space))
-					var/turf/mob_loc = get_turf(M)
-					loc_display = mob_loc.loc
-				if(T.malfunction)
-					loc_display = pick(teleportlocs)
-				dat += "ID: [T.id] | Location: [loc_display]<BR>"
-				dat += "<A href='?src=\ref[src];warn=\ref[T]'>(<font color=red><i>Message Holder</i></font>)</A> |<BR>"
-				dat += "********************************<BR>"
-			dat += "<HR><A href='?src=\ref[src];lock=1'>Lock Console</A>"
-
-		user << browse(dat, "window=computer;size=400x500")
-		onclose(user, "computer")
+/obj/machinery/computer/prisoner/attack_hand(var/mob/user)
```

Comment:
Strongly suggest
```
/obj/machinery/computer/prisoner/interact(mob/user)
(move all code here, doesn't call parent)

/obj/machinery/computer/prisoner/interface_interact(mob/user)
  interact(user)
  return TRUE
```
Same goes for all `/machinery` subtypes. This will handle all interactivity, topic, update, machine set, etc. for you.